dask_remote/runner/api.py

```diff
@@ -0,0 +1,71 @@
+import math
+from multiprocessing import Process
+
+import uvicorn
+from fastapi import FastAPI
+from pydantic import BaseModel
+
+from .cluster_process import ClusterProcessProxy
+
+
+class MessageResponse(BaseModel):
+    message: str
+
+
+def cluster_api(cluster_proxy: ClusterProcessProxy, fastapi_kwargs: dict):
+    """Create a FastAPI app that exposes given ClusterProcessProxy."""
+    app = FastAPI(**fastapi_kwargs)
+    app.cluster = cluster_proxy
+
+    @app.get("/", response_model=MessageResponse)
+    async def root():
+        return MessageResponse(message="Hello from cluster controller")
+
+    @app.get("/status", response_model=MessageResponse)
+    async def status():
+        return MessageResponse(message=app.cluster.status)
+
+    @app.get("/scheduler_address", response_model=MessageResponse)
+    async def scheduler_address():
+        return MessageResponse(message=app.cluster.scheduler_address)
+
+    @app.get("/scale", response_model=MessageResponse)
+    async def scale():
+        """Get current number of workers."""
+        return MessageResponse(message=str(app.cluster.num_workers))
+
+    @app.post("/scale/{n}", response_model=MessageResponse)
+    async def scale(n: int):
+        """Scale to `n` workers."""
+        if n < 0:
+            n = 0
+        app.cluster._adaptive_stop()
+        app.cluster.scale(n)
+        return MessageResponse(message=f"Scaling to {n} workers")
+
+    @app.post("/adapt", response_model=MessageResponse)
+    async def adapt(minimum: int = 0, maximum: int = None):
+        """Set cluster to adaptive scaling mode."""
+        if maximum is None:
+            maximum = math.inf
```

Comment:
hur

Reply:
btw why don't you set it to math.inf directly? it's not an object

Reply:
@capitancambio not sure what you mean?

Reply:
:heavy_check_mark: